### PR TITLE
Add support for nested queries

### PR DESCRIPTION
**Pull request by janhohenheim**

### CODE REVIEW

File: LinqToRest.Test/HttpClientMock.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+            return this;
+        }
+
+        public HttpClient ToHttpClient()
+        {
+            return _mockHttp.ToHttpClient();
+        }
+    }
+}
```
Comment (dngroth) [inline LinqToRest.Test/HttpClientMock.cs]: newline!

File: LinqToRest.Test/QueryProviderTest.cs [hunk 2]
```diff
@@ -13,66 +14,209 @@ public class QueryProviderTest
...
-            var serviceUri = MockServiceUri();
-            var resourceRetriever = MockResourceRetriever();
-            var queryBinderFactory = MockQueryBinderFactory();
-            var queryProvider = new QueryProvider(resourceRetriever, queryBinderFactory, serviceUri);
+            var actual = CreateQuery<EntityWithQueryableMember>()
+                .Select(entity => new { entity.Name })
+                .ToString();
 
-            return new Query<T>(queryProvider);
+            var expected = UniqueIdentifierNameResult;
```
Comment (dngroth) [inline LinqToRest.Test/QueryProviderTest.cs]: Move `.Query ` to `var expected` evrywere in file

File: LinqToRest.Test/QueryProviderTest.cs [hunk 3]
```diff
@@ -13,66 +14,209 @@ public class QueryProviderTest
...
-        private static QueryBinderFactory MockQueryBinderFactory()
+        #endregion
+
+        #region Mock
+
+        private static HttpClient MockHttpClient()
         {
-            var queryBinderFactory = Substitute.For<QueryBinderFactory>();
-            queryBinderFactory().Returns(new QueryBinder(new EntityValidator()));
+            var mockHttp = new MockHttpMessageHandler();
```
Comment (dngroth) [inline LinqToRest.Test/QueryProviderTest.cs]: use `HttpClientMock()`

File: LinqToRest.Test/ResourceRetrieverTest.cs [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+                {
+                    UniqueIdentifier = "Test2",
+                    Name = "Test2"
+                }
+            });
+        private static string NameQueryableMemberRequestUri =>
+            $"{RootUri}entitywithqueryablemembers?fields=uniqueIdentifier,name,queryableMember";
+
+        private static QueryResult<object> NameQueryableMemberResult => new QueryResult<object>(
+        new Uri(NameQueryableMemberRequestUri),
```
Comment (dngroth) [inline LinqToRest.Test/ResourceRetrieverTest.cs]: Is indentation correct?

File: LinqToRest/IResourceRetriever.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace Messerli.LinqToRest
+{
+    public interface IResourceRetriever
+    {
+        T RetrieveResource<T>(Uri uri);
+
+        object RetrieveResource(Type type, Uri uri);
+    }
+}
```
Comment (dngroth) [inline LinqToRest/IResourceRetriever.cs]: Add terminating newline

File: LinqToRest/ResourceRetriever.cs [hunk 6]
```diff
@@ -0,0 +1,127 @@
+﻿using System;
```
Comment (dngroth) [inline LinqToRest/ResourceRetriever.cs:1]: Cleanup functions?

**Review by dngroth — CHANGES_REQUESTED** [review]
And extend changelog

File: LinqToRest/IResourceRetriever.cs (hunk 5 above)
Comment (janhohenheim) [reply LinqToRest/IResourceRetriever.cs]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/f6bf1050eaeec4b5eadbc26c5304e13fa0470b0b

File: LinqToRest.Test/HttpClientMock.cs (hunk 1 above)
Comment (janhohenheim) [reply LinqToRest.Test/HttpClientMock.cs]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/f6bf1050eaeec4b5eadbc26c5304e13fa0470b0b

File: LinqToRest.Test/QueryProviderTest.cs (hunk 2 above)
Comment (janhohenheim) [reply LinqToRest.Test/QueryProviderTest.cs]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/3cfbe26f2b65048b507dff21c7b572f2d5a5ae1c

File: LinqToRest.Test/ResourceRetrieverTest.cs (hunk 4 above)
Comment (janhohenheim) [reply LinqToRest.Test/ResourceRetrieverTest.cs]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/1163b4a703b68a8db5e4d2f059e0ab8f52d917b8

File: LinqToRest.Test/QueryProviderTest.cs (hunk 3 above)
Comment (janhohenheim) [reply LinqToRest.Test/QueryProviderTest.cs]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/f09a88cdbf86a71763697a2facc065dc21ddd2e3

File: LinqToRest/IResourceRetriever.cs (hunk 5 above)
Comment (jnferner) [reply LinqToRest/IResourceRetriever.cs]: > 
> 
> And extend changelog

Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/69e58c8217304840c57d388036909a28c2f7031d and https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/4b0bc65a985451504e535fba0cfaa42e8bb518a8

**Comment by janhohenheim** [comment]
> 
> 
> And extend changelog

Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/69e58c8217304840c57d388036909a28c2f7031d and https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/4b0bc65a985451504e535fba0cfaa42e8bb518a8

File: LinqToRest/ResourceRetriever.cs (hunk 6 above)
Comment (janhohenheim) [reply LinqToRest/ResourceRetriever.cs:1]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/f597f32ca9af08d1f52582ac308e5c9c0853b2c3, https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/5b64ed9b05cb82f4254a44964b9adafeb6d43d69 and https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/49cf9a9be4fa79c4f240dff62fa65692706dc843

File: LinqToRest.Test/QueryProviderTest.cs (hunk 3 above)
Comment (dngroth) [reply LinqToRest.Test/QueryProviderTest.cs]: `retrun new HttpClientMock()
    .RegisterJasonResponse(...)
    .ToHttpClient();`

how about that?

Comment (janhohenheim) [reply LinqToRest.Test/QueryProviderTest.cs]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/ae816f8de380d69c5b5467919bd74346e54c6310

**Review by dngroth — DISMISSED** [review]

**Review by dngroth — DISMISSED** [review]

**Review by dngroth — APPROVED** [review]

**Pull request merged**